clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -207,6 +208,84 @@ public void testCreateTopics() throws Exception {
         }
     }
 
+    @Test
+    public void testAdminClientApisWithAuthenticationFailure() throws Exception {
+        MockKafkaAdminClientEnv env = mockClientEnv();
+        Node node = env.cluster().controller();
+        env.kafkaClient().setNodeApiVersions(NodeApiVersions.create());
+        env.kafkaClient().setNode(env.cluster().controller());
+        env.kafkaClient().authenticationFailed(env.cluster().controller(), 300);
+
+        callAdminClientApisAndExpectAnAuthenticationError(env);
+
+        env.kafkaClient().authenticationSucceeded(node);
+        env.time().sleep(30); // wait less than retry backoff period
+
+        callAdminClientApisAndExpectAnAuthenticationError(env);
+
+        env.time().sleep(300); // wait long enough this time
+
+        // try one of the apis to verify authentication error is gone
+        env.kafkaClient().prepareMetadataUpdate(env.cluster(), Collections.<String>emptySet());
+        env.kafkaClient().prepareResponse(new CreateTopicsResponse(Collections.singletonMap("myTopic", new ApiError(Errors.NONE, ""))));
+        env.adminClient().createTopics(
+                Collections.singleton(new NewTopic("myTopic", Collections.singletonMap(Integer.valueOf(0), asList(new Integer[]{0, 1, 2})))),
+                new CreateTopicsOptions().timeoutMs(10000)).all().get();
+
+        env.close();
+    }
+
+    private void callAdminClientApisAndExpectAnAuthenticationError(MockKafkaAdminClientEnv env) throws InterruptedException {
+        env.kafkaClient().prepareMetadataUpdate(env.cluster(), Collections.<String>emptySet());
+
+        try {
+            env.adminClient().createTopics(
```

Comment:
Would it make sense to use different test cases for each of these calls? Is there significant overhead we're saving by combining them?